site/content/entry/citra-progress-report-2017-july.md

```diff
@@ -0,0 +1,62 @@
++++
+date = "2017-08-01T12:38:00-04:00"
+title = "Citra Progress Report - 2017 July"
+tags = [ "progress-report" ]
+author = "Saphiresurf"
+forum = 0000
++++
+
+With June bringing in many huge improvements to GPU functionality, July has come in to sweep up and improve accuracy in a few distinct areas of Citra. Changes and fixes ranging from texture handling all the way to aesthetic changes that help improve the visual appeal and customizability of Citra's QT interface. Now, let's get into what's changed!
+<br />
+
+## [GPU: Fix Edge Cases for TextureCopy](https://github.com/citra-emu/citra/pull/2809) by [wwylele](https://github.com/wwylele)
+
+What texture copy essentially does is perform a copy of a texture, but with a configurable gap in case the texture is going to be copied into a smaller resolution area. In a bit more technical terms TextureCopy is a data transfer mode handled by the GPU that's activated when the third bit in the GPU's flags register (a register which is only accessible by the ARM11 processor) is set to 1. This particular process ignores all other bits in the gpu's flags register except for bit 2 (which still needs to be set so the GPU knows if it needs to crop the texture or not) and regular dimension registers are ignored.
```

Comment:
```diff
@@ 14,1 2017,08,13,17,51 @@
-- What texture copy essentially does is perform a copy of a texture, but with a configurable gap in case the texture is going to be copied into a smaller resolution area. In a bit more technical terms TextureCopy is a data transfer mode handled by the GPU that's activated when the third bit in the GPU's flags register (a register which is only accessible by the ARM11 processor) is set to 1
+++The 3DS GPU has a data transfer mode called `TextureCopy`, which as the name says, is for copying textures but with a configurable gap, in case the texture is going to be copied into a smaller resolution area. This mode is enabled by setting the 3rd flag in the GPU and causes it to ignore every other flag except the 2nd, which is used to tell it whether or not to crop the texture. 
```